### ISSUE TITLE
[DI] Optimize koin dependency declarations

### ISSUE THREAD

**Issue by y9vad9** [timemates/backend#137]
For now, there's a lot of complex code that can be reduced by using ['constructor dsl'](https://insert-koin.io/docs/reference/koin-core/dsl-update/) in koin.

You should optimize next code (it's just example):
```kotlin
single {
        ConfigureNewAccountUseCase(
            users = get(),
            authorizations = get(),
            verifications = get(),
            timeProvider = get(),
            randomProvider = get()
        )
    }
```
To:
```kotlin
singleOf(::ConfigureNewAccountUseCase)
```

It's possible becase we have already declared all other dependencies.

**Comment by y9vad9** [comment]
also, you can optimize singles that doesn't have any parameters, like this:
```kotlin
single {
    AuthorizationsMapper()
}
```
You can optimize to next:
```kotlin
singleOf(::AuthorizationsMapper)
```

**Comment by y9vad9** [comment]
closed within #147
